plugins/logitech-bulkcontroller/fu-plugin-logitech-bulkcontroller.c

```diff
@@ -0,0 +1,388 @@
+					  read_upd_data_cb,
+					  read_sync_data_cb,
+					  send_data_sync_cb};
+
+	plugin_data = fu_plugin_get_data(plugin);
+	plugin_data->bulkcontroller_obj = NULL;
+	/* Check if Logitech Rally Bar Mini is conneced */
+	pid = 0x08D3;
+	plugin_data->device_name = g_strdup("Rally Bar Mini");
+	plugin_data->device_type = kDeviceTypeRallyBarMini;
+	plugin_data->device_info_available = FALSE;
+	plugin_data->run_devicemode_info_ = FALSE;
+	obj = logibulkcontroller_create_bulk_controller(0x46d, pid, bulkcb, plugin_data);
+	if (ERRORCODE_NO_ERROR == logibulkcontroller_open_device(obj)) {
+		g_autoptr(FuDevice) device = NULL;
+		plugin_data->bulkcontroller_obj = obj;
+		if (!send_device_info_request(plugin_data)) {
+			g_warning(" Failed to communicate with device %s",
+				  plugin_data->device_name);
+			logibulkcontroller_close_device(obj);
+			return FALSE;
+		}
+		/* Wait for the response */
+		sleep(TIMEOUT_SYNC_RESPONSE_5SEC);
+		if (!plugin_data->device_info_available) {
+			g_warning(" Failed to receive device information from device %s",
+				  plugin_data->device_name);
+			logibulkcontroller_close_device(obj);
+			return FALSE;
+		}
+		ctx = fu_plugin_get_context(plugin);
+
+		device = fu_device_new_with_context(ctx);
+		fu_device_set_id(device, "logibulkctrl-video-conf-devices");
+		fu_device_add_guid(device, "7da607b9-1f8f-597a-8f5e-5788d6f8bcb3");
+		fu_device_set_name(device, plugin_data->device_info.name);
+		fu_device_set_version_format(device, FWUPD_VERSION_FORMAT_TRIPLET);
+		fu_device_set_vendor(device, "Logitech");
+		fu_device_add_vendor_id(device, "USB:0x046D");
+		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE);
+		fu_device_add_protocol(device, "com.logitech.vc.proto");
+		fu_device_set_version(device, plugin_data->device_info.sw);
+		fu_plugin_device_add(plugin, device);
+		return TRUE;
+	}
+	/* Check if Logitech Rally Bar is conneced */
+	pid = 0x089B;
+	plugin_data->device_name = g_strdup("Rally Bar");
+	plugin_data->device_type = kDeviceTypeRallyBar;
+	plugin_data->device_info_available = FALSE;
+	plugin_data->run_devicemode_info_ = FALSE;
+	obj = logibulkcontroller_create_bulk_controller(0x46d, pid, bulkcb, plugin_data);
+	if (ERRORCODE_NO_ERROR == logibulkcontroller_open_device(obj)) {
+		g_autoptr(FuDevice) device = NULL;
+		plugin_data->bulkcontroller_obj = obj;
+		if (!send_device_info_request(plugin_data)) {
+			g_warning(" Failed to communicate with device %s",
+				  plugin_data->device_name);
+			logibulkcontroller_close_device(obj);
+			return FALSE;
+		}
+		/* Wait for the response */
+		sleep(TIMEOUT_SYNC_RESPONSE_5SEC);
+		if (!plugin_data->device_info_available) {
+			g_warning(" Failed to receive device information from device %s",
+				  plugin_data->device_name);
+			logibulkcontroller_close_device(obj);
+			return FALSE;
+		}
+		ctx = fu_plugin_get_context(plugin);
+
+		device = fu_device_new_with_context(ctx);
+		fu_device_set_id(device, "logibulkctrl-video-conf-devices");
```

Comment:
I don't think you want to be using the same device ID for both types of devices - what if they're both connected at the same time?